pkg/infra/http.go

```diff
@@ -8,7 +8,24 @@ import (
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
 )
 
-func HTTPTracedTransport(rt http.RoundTripper, serviceName string) http.RoundTripper {
+type beforeFn func(req *http.Request, span ddtrace.Span)
+type config struct {
+	before beforeFn
+}
+type OptionFn func(c *config)
+
+func WithBefore(before beforeFn) OptionFn {
```

Comment:
might want to namespace this a bit further seeing as we're in a shared infra package